lib/ex_doc.ex

```diff
@@ -25,6 +25,7 @@ defmodule ExDoc do
       project: nil,
       retriever: ExDoc.Retriever,
       source_beam: nil,
```

Comment:
> If the field can be an empty string and an empty string also mean "emptiness", we need to convert the empty string to nil before putting it in the struct.

Yes, we should preprocess those empty strings and convert the data to `nil`. Also, it looks better doing this:

```elixir
if options.string_field do
  # ...
end
```

than this:

```elixir
if  options.string_field != "" do
  # ...
end
```

:)